mlrun/pipelines/__init__.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+from importlib.abc import MetaPathFinder
+from importlib.util import spec_from_file_location
+
+import kfp
+
+
+def resolve_pipeline_engine():
+    if kfp.__version__.startswith("2.0"):
+        return "kfp-v2.0"
+    return "kfp-v1.8"
+
+
+PIPELINE_COMPATIBILITY_MODE = resolve_pipeline_engine()
+_pipeline_module_locations = {
+    "kfp-v1.8": "mlrun/pipelines/kfp/v1_8",
+    "kfp-v2.0": "mlrun/pipelines/kfp/v2_0",
+}
+
+
+class PipelineEngineModuleFinder(MetaPathFinder):
+    @staticmethod
+    def _resolve_module_path(fullname, path):
+        path_prefix = path[0].replace(
+            "mlrun/pipelines",
+            _pipeline_module_locations.get(PIPELINE_COMPATIBILITY_MODE),
+        )
+        path_suffix = fullname.replace("mlrun.pipelines", "")
```

Comment:
From a somewhat surface-level read of the path-finder mechanism, I'm not sure how this will work with imports of relative paths? Specifically if importing `..pipelines.something` - this `replace` will not do anything, and it seems that your next replace (`.`->`/`) will change this to something like `//pipelines/something` which doesn't seem to be what we want? You need to replace `..` to `../`, don't you?

Reply:
This works with relative imports. Before calling the finders, Python's import system completes the import string.

However, based on your and Liran's feedback, I've changed this import strategy to one that favors static analyzers such as `lint-imports` and PyCharm's autocomplete. It's not completely ready yet, but you can already take a look and give me your thoughts (I've pushed it to this PR).